Sources/SwiftLanguageService/CodeCompletionSession.swift

```diff
@@ -704,6 +719,29 @@ class CodeCompletionSession {
 
     return Position(line: completionPos.line, utf16index: deletionStartUtf16Offset)
   }
+
+  /// Generate a snippet for control flow keywords like if, for, while, etc.
+  /// Returns the snippet text if the keyword is a control flow keyword and snippets are supported, otherwise nil.
+  private func keywordSnippet(for keyword: String) -> String? {
+    guard clientSupportsSnippets else { return nil }
+
+    switch keyword {
+    case "if":
+      return "if ${1:condition} {\n\t${0:}\n}"
```

Comment:
We should not assume that the source file is indented with tabs. We have `BasicFormat.inferIndenatation` to infer a source files’s indentation and should rely and that instead.